### PR TITLE
Combine test image concepts of os, flavor and arch

### PR DESCRIPTION
**Pull request by stefwalter**
We don't need separate concepts for all of these, just a single
set of image names.

For now we hardcode the fact that all the images are x86_64
for the few cases where that information is required, but in
the future we can easily have a list of which images are
from different architectures.

### CODE REVIEW

**Comment by stefwalter** [comment]
Blocked on #3208

**Comment by stefwalter** [comment]
Rebased

File: no-machine.txt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+x bootstrap script no longer gets argument
```
Comment (mvollmer) [inline no-machine.txt]: This file shouldn't be here anymore, right?

Comment (stefwalter) [inline no-machine.txt]: Yup, will remove.

**Comment by stefwalter** [comment]
This is a prelude to my work on having a verifier machine test any OS.

**Pull request closed without merging**